tests/test_schema_retrieval.py

```diff
@@ -26,29 +26,34 @@
 class TestSchemaRetrieval:
     "Test schema retrieval."
 
+    @pytest.mark.xfail(reason="default remote might be down but it's not this library's issue", raises=ConnectionError)
```

Comment:
I was testing this for like 20 min trying to get it to XFAIL with my wifi turned off but it kept failing regularly. I think what you need to do is use the `ConnectionError` from `requests`.

e.g.

```python
from requests.exceptions import ConnectionError
from pydax._schema_retrieval import retrieve_schema_file

try:
    retrieve_schema_file('https://www.rgjbweibfeuwbfuiwebfuwbfuiebiubfw.com')
except ConnectionError:
    print('Caught the exception')
except Exception as ex:
    print('Didnt catch the exception')
    print(ex.__class__.__name__)
    print(ex)
```

If you comment out the first line, it won't get caught by `except ConnectionError`

Reply:
Great catch, pushed the fix. I just assumed the `ConnectionError` I got was the builtin one.

I added `requests.ConnectionError` to avoid name clash with the builtin `ConnectionError`.

Reply:
makes sense, pretty soon we will need tests for testing our tests 😆

Reply:
I moved this test to `test_schema.py` and added some other changes, because this is more about testing the content of the schema URLs rather than the retrieving functionality